### PR TITLE
Added uuid resolve for Wordpress CPHs.

### PR DESCRIPTION
**Pull request by bogdanguranda**

### CODE REVIEW

**Comment by kovacshuni** [comment]
I think mostly it's ok though:
* i think if you could avoid modifying the interface at `Initialize()`, affecting all implementing structs that would be better.
* and creating an http client every time one calls the function `resolveUUID()` and creating the `NewHttpDocStoreClient` is not optimal either. You could try injecting things better or even creating a separate entity alongside `methodeContent.go`

but i'll take a look tomorrow again.

**Comment by bogdanguranda** [comment]
@kovacshuni I totally agree, this is just a first draft so that I can test this a bit that it works

**Comment by kovacshuni** [comment]
all ✅  except the failing circle build

File: app.go [hunk 1]
```diff
@@ -8,20 +8,21 @@ import (
 	"os"
```
Comment (MihaiSorinMoisa) [inline app.go]: Organzie imports the following way https://github.com/golang/go/wiki/CodeReviewComments#imports.

File: checks/docStoreClient.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package checks
+
+import (
```
Comment (MihaiSorinMoisa) [inline checks/docStoreClient.go]: Organize imports the following way https://github.com/golang/go/wiki/CodeReviewComments#imports.

File: checks/docStoreClient.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	resp, err := c.httpCaller.DoCall(Config{
+		Url:      docStoreUrl.String(),
+		Username: c.username,
+		Password: c.password,
+		TxId:     ConstructPamTxId(tid),
+	})
+
+	if err != nil {
+		return -1, "", fmt.Errorf("Unsucessful request for fetching canonical identifier for authority=%v identifier=%v url=%v, error was: %v", authority, identifier, docStoreUrl.String(), err.Error())
```
Comment (MihaiSorinMoisa) [inline checks/docStoreClient.go]: Fix typo in `Unsucessful`.

File: methodeTypeResolver.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	attributes, err := m.buildAttributes(eomFile.Attributes)
+	if err != nil {
+		return "", err
+	}
+
+	uuid := ""
+	if isBlogCategory(attributes) {
+		resolvedUuid, err := m.iResolver.ResolveIdentifier(attributes.ServiceId, attributes.RefField, txID)
+		if err != nil {
+			return "", fmt.Errorf("Couldn't resolve blog uuid %v", err)
```
Comment (MihaiSorinMoisa) [inline methodeTypeResolver.go]: Add a colon after uuid to delimit the start of the inner error message.

File: methodeTypeResolver.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+			return "", "", err
+		}
+
+		theType := "EOM::CompoundStory_External_CPH"
+		cphUuid := eomFile.UUID
+		if resolvedUuid != "" {
+			theType = "EOM::CompoundStory_Internal_CPH"
+			cphUuid = resolvedUuid
+		}
+		log.Infof("For placeholder resolved type=%v uuid=%v", theType, cphUuid)
```
Comment (MihaiSorinMoisa) [inline methodeTypeResolver.go]: Add txID to the log message as well.

File: content/content.go [hunk 6]
```diff
@@ -23,6 +17,9 @@ type Content interface {
 	GetUUID() string
 }
 
+type InitConfig struct {
```
Comment (MihaiSorinMoisa) [inline content/content.go]: This type doesn't seem to be used anywhere.

**Comment by MihaiSorinMoisa** [comment]
Don't forget to run the command `govendor fmt +local`

**Pull request merged**